src/batch/src/executor2/generate_series.rs

```diff
@@ -12,49 +12,110 @@
+use risingwave_expr::expr::build_from_prost;
 use risingwave_pb::batch_plan::plan_node::NodeBody;
 
 use crate::executor::ExecutorBuilder;
-use crate::executor2::{BoxedDataChunkStream, BoxedExecutor2, BoxedExecutor2Builder, Executor2};
-
-pub struct GenerateSeriesI32Executor2 {
-    start: i32,
-    stop: i32,
-    step: i32,
-    cur: i32, // Current value in the series.
+use crate::executor2::{
+    BoxedDataChunkStream, BoxedExecutor2, BoxedExecutor2Builder, Executor2,
+    GenerateSeriesI32Executor2, GenerateSeriesTimestampExecutor2,
+};
 
+pub struct GenerateSeriesExecutor2Wrapper {
+    input: BoxedExecutor2,
     schema: Schema,
     identity: String,
 }
 
-impl BoxedExecutor2Builder for GenerateSeriesI32Executor2 {
+impl BoxedExecutor2Builder for GenerateSeriesExecutor2Wrapper {
     fn new_boxed_executor2(source: &ExecutorBuilder) -> Result<BoxedExecutor2> {
         let node = try_match_expand!(
             source.plan_node().get_node_body().unwrap(),
-            NodeBody::GenerateInt32Series
+            NodeBody::GenerateSeries
         )?;
 
-        Ok(Box::new(Self {
-            start: node.start,
-            stop: node.stop,
-            step: node.step,
-            cur: node.start,
-            schema: Schema::new(vec![Field::unnamed(DataType::Int32)]),
-            identity: source.plan_node().get_identity().clone(),
-        }))
+        let identity = source.plan_node().get_identity().clone();
+
+        let dummy_chunk = DataChunk::new_dummy(1);
+        let start_expr = build_from_prost(node.get_start()?)?;
+        let stop_expr = build_from_prost(node.get_stop()?)?;
+        let step_expr = build_from_prost(node.get_step()?)?;
+
+        let start = start_expr.eval(&dummy_chunk)?;
+        let stop = stop_expr.eval(&dummy_chunk)?;
+        let step = step_expr.eval(&dummy_chunk)?;
+
+        match start_expr.return_type() {
+            DataType::Timestamp => {
+                let start = start.as_naivedatetime().value_at(0);
+                let stop = stop.as_naivedatetime().value_at(0);
+                let step = step.as_interval().value_at(0);
+
+                if let (Some(start), Some(stop), Some(step)) = (start, stop, step) {
+                    let schema = Schema::new(vec![Field::unnamed(DataType::Timestamp)]);
+
+                    let input = Box::new(GenerateSeriesTimestampExecutor2::new(
+                        start,
+                        stop,
+                        step,
+                        schema.clone(),
+                        identity.clone(),
+                    ));
+
+                    Ok(Box::new(Self {
+                        input,
+                        schema,
+                        identity,
+                    }))
+                } else {
+                    Err(ErrorCode::InternalError(
+                        "the parameters of Generate Series Function are incorrect".to_string(),
+                    )
+                    .into())
```

Comment:
FYI PostgreSQL returns 0 rows rather than reporting error when any of the 3 args is null. Can fix it later.